src/main/java/com/google/sps/data/BusinessesService.java

```diff
@@ -150,109 +128,156 @@ private void addListingToBusinesses(PlacesSearchResult place, String url) {
+          checkBusinessThroughLinkedin(currentBusiness)) {
+      addBusinessToDatabase(currentBusiness, BIG_BUSINESSES_DATABASE);
+      return true;
+    }
+
+    addBusinessToDatabase(currentBusiness, SMALL_BUSINESSES_DATABASE);
+    return false;
+  }
+
+  private boolean checkIfBusinessInDatabase(Listing currentBusiness, 
+                                              String databaseEntry) {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();    
+    
+    try {
+        String businessName = 
+            (String) datastore.get(KeyFactory.createKey(
+                                    databaseEntry, 
+                                    currentBusiness.getName()))
+                              .getProperty("Business");
+        return true;
+    } catch (EntityNotFoundException e) {
+        // Sometimes the small or big business will be in the database,
+        // but if not additional checks will follow to determine it's size.
+        LOGGER.warning(e.getMessage());
+        return false;
+    }
+  }
+
+  private boolean checkNumberOfSimilarBusinesses(Listing currentBusiness) {
+    GeoApiContext context = 
+      new GeoApiContext.Builder().apiKey(KEY).build();
+    TextSearchRequest request = 
         new TextSearchRequest(context).query(currentBusiness.getName())
           .location(latLng).radius(50000);
+    PlacesSearchResult[] similarBusinessesInTheArea = null;
 
-      try {                                       
-        PlacesSearchResult[] similarBusinessesInTheArea = 
-          request.await().results;
+    try {                                       
+         similarBusinessesInTheArea = request.await().results;
+    } catch(IOException | InterruptedException | ApiException e ) {
+        LOGGER.warning(e.getMessage());
+    } 
 
-        if (similarBusinessesInTheArea.length > 1) {
-          // Quickly determines if a business is big, instead of having to iterate
-          // through all the places that are similar to it in the area. Also,
-          // useful if a business does not have that many loacations opened, 
-          // but has a huge following on linkedin.
-          checkBusinessThroughLinkedin(currentBusiness, similarBusinessesInTheArea);
-        }else if(similarBusinessesInTheArea.length == 1) {
-          numberOfSmallBusinesses++;
-        }
-      } catch(GeneralSecurityException | IOException | InterruptedException | ApiException e ) {
-          LOGGER.warning(e.getMessage());
-        } 
+    return checkNumberOfSimilarBusinessesInTheArea(currentBusiness, 
+                                                  similarBusinessesInTheArea);  
+  }
+
+  private boolean checkNumberOfSimilarBusinessesInTheArea(Listing currentBusiness, 
+                          PlacesSearchResult[] similarBusinessesInTheArea) {
+    int numberOfMatchingBusinesses = 0;
+    int i = 0;
+
+    while (i < similarBusinessesInTheArea.length 
```

Comment:
Thanks for moving these methods right after each other!

If the try on line 194/195 fails, I think this will through a null error. A safe resolution would be to initialize similarBusinessesInTheArea to an empty list instead. Then if the try fails, the while loop never executes and you return false. This seems appropriate given there would be no similar businesses to consider.

Reply:
Yea I forgot to change it to an empty list, but it should be good now